### PR TITLE
feat: Add `ordered_types` to `@Symfony`

### PR DESCRIPTION
**Pull request by VincentLanglet**
Hi @Wirone 

Symfony coding style already use:
```
'phpdoc_types_order' => [
                'null_adjustment' => 'always_last',
                'sort_algorithm' => 'none',
            ]
```
so I assume same should be applied to the types, ie:
```
'ordered_types' => [
                'null_adjustment' => 'always_last',
                'sort_algorithm' => 'none',
            ]
```

### CODE REVIEW

**Comment by SpacePossum** [comment]
The failures are a side effect of your changes.
The test `tests/Fixtures/Integration/misc/PHP8_2.test` is run with the configuration `@PhpCsFixer` which contains/extends itself the Symfony set. Since you changed the SF set the new configuration is applied on the `tests/Fixtures/Integration/misc/PHP8_2.test`.

The diff now looks like:

```diff
-function falsyNull(null $n, false $f): null|false {}
+function falsyNull(null $n, false $f): false|null {}
```

Now this expected since you've changed the configuration, however you didn't update the expected output for the test, hence it fails. So the fix would be to update the expected output for the test file and all should be fine :)

**Comment by VincentLanglet** [comment]
Thanks

**Comment by SpacePossum** [comment]
Might be worth running on latest SF src and see how many hits there are and/or align with SF team, anyways, change look good to me, so I would welcome these :)

**Comment by VincentLanglet** [comment]
> Might be worth running on latest SF src and see how many hits there are and/or align with SF team, anyways, change look good to me, so I would welcome these :)

I found 8 occurences of `null|` and tons of `|null`

**Comment by SpacePossum** [comment]
thanks, sounds like a good thing you got here :)

**Comment by Wirone** [comment]
@nicolas-grekas can you confirm we can safely add it? @VincentLanglet did you try to run it against Symfony codebase?

File: src/RuleSet/Sets/SymfonySet.php [hunk 1]
```diff
@@ -147,6 +147,10 @@ public function getRules(): array
                 ],
                 'sort_algorithm' => 'alpha',
             ],
+            'ordered_types' => [
+                'null_adjustment' => 'always_last',
+                'sort_algorithm' => 'none',
```
Comment (keradus) [inline src/RuleSet/Sets/SymfonySet.php:152]: can't we use any sorting here?

Comment (keradus) [reply src/RuleSet/Sets/SymfonySet.php:152]: regardless of decision of the Sf, we can at least do it for us.
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7361

Comment (VincentLanglet) [reply src/RuleSet/Sets/SymfonySet.php:152]: Since they use 
```
'phpdoc_types_order' => [
     'null_adjustment' => 'always_last',
     'sort_algorithm' => 'none',
],
```
I don't see why they would sort the types and not the phpdoc.

But indeed PhpCsFixerSet can use `ordered_type => true`

Comment (keradus) [reply src/RuleSet/Sets/SymfonySet.php:152]: let me rephrase my original question - wouldn't Symfony as community be interested to sort in both cases?

Comment (VincentLanglet) [reply src/RuleSet/Sets/SymfonySet.php:152]: Dunno 🤷‍♂️
I only know about their standard here: https://symfony.com/doc/current/contributing/code/standards.html

**Comment by keradus** [comment]
@VincentLanglet , no need to poke me for review.
the missing element here is voice of Symfony to declare if they want to extend SymfonySet this way , eg https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7356#issuecomment-1748511448

**Comment by VincentLanglet** [comment]
Maybe @OskarStark you could say if it's ok to add this to the Symfony standard or ping someone who could decide

**Comment by OskarStark** [comment]
I will bring this PR to the core-Team, let's wait a decision.

**Comment by VincentLanglet** [comment]
> I will bring this PR to the core-Team, let's wait a decision.

Any news @OskarStark ? ^^'

**Review by keradus — COMMENTED** [review]
Accepted by SF community

File: src/RuleSet/Sets/SymfonySet.php [hunk 2]
```diff
@@ -147,6 +147,10 @@ public function getRules(): array
                 ],
                 'sort_algorithm' => 'alpha',
             ],
+            'ordered_types' => [
```
Comment (keradus) [inline src/RuleSet/Sets/SymfonySet.php:150]: would be good to cover this set change in test.
ie tests/Fixtures/Integration/set/@Symfony.test

Comment (VincentLanglet) [reply src/RuleSet/Sets/SymfonySet.php:150]: It won't be compatible with the fact the test is running in PHP 7.4

Comment (VincentLanglet) [reply src/RuleSet/Sets/SymfonySet.php:150]: @keradus I don't see how to add this in Symfony.set since it must run in PHP 7.4 too.

But there is already test about this change since I needed to update the following code:
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7356#issuecomment-1747299086

**Comment by VincentLanglet** [comment]
Test are green and I don't think https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7356#discussion_r1453123282 is doable until we drop 7.4 support.

Is it something else to do @Wirone @keradus ? :)
